environ/src/environ/core.clj

```diff
@@ -36,10 +36,23 @@
       (into {} (for [[k v] (edn/read-string (slurp env-file))]
                  [(sanitize-key k) (sanitize-val k v)])))))
 
+(defn- warn-on-overwrite [ms]
+  (doseq [[k kvs] (group-by key (apply concat ms))
+          :let [vs (map val kvs)
```

Comment:
Can you align the clauses? i.e.

```clojure
  (doseq [[k kvs] (group-by key (apply concat ms))
          :let    [vs      (map val kvs)
                   initial (first vs)
                   final   (last vs)]
          :when   (and (next kvs) (not= initial final))]
```

It makes the bindings and values a little clearer.

Also, I wonder if we need `initial` and `final`. I think it might be better written as:

```clojure
(defn- warn-on-overwrite [ms]
  (doseq [[k kvs] (group-by key (apply concat ms))
          :let    [vs (map val kvs)]
          :when   (and (next kvs) (not= (first vs) (last vs)))]
    (println "Warning: environ value" (first vs) "for key" k
             "has been overwritten with" (last vs))))